src/background/messenger.ts

```diff
@@ -19,36 +19,45 @@ export interface ExtensionAdapter {
 }
 
 export default class Messenger {
-    private reporters: Set<(info: ExtensionData) => void>;
     private adapter: ExtensionAdapter;
+    private changeListenerCount: number;
 
     constructor(adapter: ExtensionAdapter) {
-        this.reporters = new Set();
         this.adapter = adapter;
-        chrome.runtime.onConnect.addListener((port) => {
-            if (port.name === 'ui') {
-                port.onMessage.addListener(async (message) => await this.onUIMessage(port, message));
+        this.changeListenerCount = 0;
+        chrome.runtime.onMessage.addListener((message: Message, sender: any, sendResponse: (any) => void) => {
+            if (message.name === 'ui') {
+                this.onUIMessage(message, sendResponse);
                 this.adapter.onPopupOpen();
+                return ([
+                    'get-data',
+                    'get-active-tab-info',
+                    'apply-dev-dynamic-theme-fixes',
+                    'apply-dev-inversion-fixes',
+                    'apply-dev-static-themes'
+                ].includes(message.type));
             }
         });
     }
 
-    private async onUIMessage(port: chrome.runtime.Port, {type, id, data}: Message) {
+    private async onUIMessage({type, id, data}: Message, sendResponse: (any) => void) {
```

Comment:
```suggestion
    private async onUIMessage({type, data}: Message, sendResponse: (any) => void) {
```
Pretty must what ESLint says. It's unused.